schemas/engagementSchema.js

```diff
@@ -1,16 +1,27 @@
 import * as yup from 'yup';
+const { ObjectId } = require('mongodb');
+
+function objectIdTest(name, isOptional) {
+  return (isOptional ? {
+    name: name,
+    message: (name + " is not a valid ObjectId"),
+    test: val => val === undefined || ObjectId.isValid(val)
+  } : {
+    name: name,
+    message: (name + " is not a valid ObjectId"),
+    test: val => ObjectId.isValid(val)
+  })
+}
```

Comment:
Since this function is in every schema, why can't we export it somewhere and then each schema can just import it?